src/gui/toolbarMenubar/model/ColorPalette.h

```diff
@@ -0,0 +1,157 @@
+#pragma once
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include <filesystem.h>
+#include <util/NamedColor.h>
+
+/**
+ * @brief Header entity of a gpt file
+ * This is simply a key value pair. The main reason to define a struct for this is to define a custom input operator.
+ *
+ */
+struct Header {
+    /**
+     * @brief Input operator for parsing Header from input string stream
+     *
+     * @param str input string stream
+     * @param header output for parsed Header
+     * @return std::istream& which is the rest of the input string stream after parsing
+     */
+    friend std::istream& operator>>(std::istream& str, Header& header);
+
+private:
+    friend class Palette;
+    /**
+     * @brief Attribute name given in gpt file
+     *
+     */
+    std::string attribute;
+
+    /**
+     * @brief Value for corresponding attribute in gpt file
+     *
+     */
+    std::string value;
+
+    /**
+     * @brief Swap all properties between two Header instances
+     *
+     * @param other other Header instance to swap properties with
+     */
+    void swap(Header& other);
+};
```

Comment:
If this struct is only used in on CU I would move it into the source file.

```suggestion
struct Header {

    friend std::istream& operator>>(std::istream& str, Header& header);

    std::string attribute; ///< Attribute name given in gpt file
    std::string value; ///< Value for corresponding attribute in gpt file
};
```

Reply:
I see your point. I'd still tend to keep it in the Header file for now. I think at some point the Header should also be used, for example for showing information about the currently active palette.
I'm not sure if I'll implement this in this PR (probably not yet) but it should still come. Then the Header class would need to be public anyways, right?